### PR TITLE
Automatically upgrade StateValueMetadata

### PR DESCRIPTION
**Pull request by msmouse**

### CODE REVIEW

**Comment by msmouse** [comment]
Current dependencies on/for this PR:
* main
  * **PR #11244** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11244?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #11262** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11262?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #11278** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11278?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #11279** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11279?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #11295** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11295?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11295?utm_source=stack-comment).

File: aptos-move/aptos-vm-types/src/abstract_write_op.rs [hunk 1]
```diff
@@ -89,7 +86,10 @@ impl AbstractResourceWriteOp {
                 metadata_op: write_op,
                 ..
             }) => write_op.get_metadata_mut(),
-            InPlaceDelayedFieldChange(_) | ResourceGroupInPlaceDelayedFieldChange(_) => None,
+            InPlaceDelayedFieldChange(_) | ResourceGroupInPlaceDelayedFieldChange(_) => {
+                // FIXME(aldenhu): confirm this is correct
+                unreachable!("No metadata for delayed field changes")
```
Comment (igor-aptos) [inline aptos-move/aptos-vm-types/src/abstract_write_op.rs:91]: we are iterating over all writes, this is definitely reachable - that's why it was an option?

"InPlaceDelayedFieldChange" means we are doing a write to that state item, it is just that only write is from Aggregator inside it, there are no other updates. 

and so the actual io write needs to be charged, it's just that state will stay the same size

**Review by igor-aptos — CHANGES_REQUESTED** [review]
still need to look at the full PR, but this needs to be fixed:

File: aptos-move/aptos-vm-types/src/abstract_write_op.rs (hunk 1 above)
Comment (msmouse) [reply aptos-move/aptos-vm-types/src/abstract_write_op.rs:95]: As discussed, made the change to let `InPlaceDelayedFieldChange` carry the metadata as well.

**Pull request merged**